### PR TITLE
Enable overriding dict type fields with null

### PR DESCRIPTION
**Pull request by glowka**
### Current behavior

Currently, `kubepy` does not support merging configurations when initial configuration defines field as dictionary and another configuration is intended to unset the field, e.g. by setting null.  In such case `kubepy` hits fatal error, because it tries to iterate over the keys of the field, no matter what's the value from overriding configuration.

### Intended behavior

So let's say, we have `env` item with `valueFrom` dictionary  field:

***deploy/example_deployment.yml***
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: example-deployment
spec:
  template:
    spec:
      containers:
      - name: example-container
        image: debian
        env:
        - name: EXAMPLE_SECRET
          valueFrom:
            secretKeyRef:
              name: example-secerts
              key: example-field
```

Now, this would be the way to unset `valueFrom` and set `value` instead:

***development_overrides/example_deployment.yml***
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: test-deployment
spec:
  template:
    spec:
      containers:
      - name: example-container
        env:
        - name: EXAMPLE_SECRET
          valueFrom: null
          value: example-dev-value
```

### Consistency with related tools/solutions 

Proposed feature is consistent with the way [helm merges values](https://github.com/helm/helm/blob/v2.16.0/docs/chart_template_guide/values_files.md#deleting-a-default-key).

---
This change is [Reviewable](https://reviewable.io/reviews/socialwifi/kubepy/36)

### CODE REVIEW

**Comment by glowka** [comment]
Hey @jgoclawski  @Behoston could you take a look here?

File: kubepy/api.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def replace(definition, namespace=None):
 
 def rolling_update(definition, name, namespace=None):
     command = kubectl_command_builder('rolling-update', name=name, namespace=namespace, with_definition=True)
-    run_command_with_yaml_on_stdin(command,definition)
+    run_command_with_yaml_on_stdin(command, definition)
```
Comment (pguz) [inline kubepy/api.py]: Let's do the refactor in the separate commit.

File: kubepy/definition_merger.py [hunk 2]
```diff
@@ -11,10 +11,13 @@ def merge_definitions(*definitions):
...
 def merge_dicts(*definitions):
-    keys = set(itertools.chain(*(definition.keys() for definition in definitions)))
+    if definitions[-1] is None:
+        return None
+    keys = set(itertools.chain(*(definition.keys() for definition in definitions if definition is not None)))
     result = {}
     for key in keys:
-        result[key] = merge_definitions(*(definition[key] for definition in definitions if key in definition))
+        result[key] = merge_definitions(*(definition[key] for definition in definitions
+                                          if definition is not None and key in definition))
```
Comment (pguz) [inline kubepy/definition_merger.py]: @glowka, as far as I see, in the current implementation key will not be unset. The `null` value will be set to the given key.
In order to unset key something like:
```suggestion
        value = merge_definitions(*(definition[key] for definition in definitions
                                          if definition is not None and key in definition))
        if value is not None:
            result[key] = value
```

should be written.

**Review by pguz — COMMENTED** [review]
Hi @glowka,

Thank you for your contribution. Please review my comments.

File: kubepy/definition_merger.py (hunk 2 above)
Comment (glowka) [reply kubepy/definition_merger.py]: The `null` value wasn't very problematic, but your solution is definitely cleaner. Done!

File: kubepy/api.py (hunk 1 above)
Comment (glowka) [reply kubepy/api.py]: OK, I removed the changes completely, assuming #35 will be merged prior to this one.

**Review by pguz — APPROVED** [review]
